pkg/app/pipedv1/plugin/kubernetes/deployment/determine.go

```diff
@@ -219,3 +221,77 @@ func checkReplicasChange(ns diff.Nodes) (before, after string, changed bool) {
 
 	return node.StringX(), node.StringY(), true
 }
+
+// First up, checks to see if the workload's `spec.template` has been changed,
+// and then checks if the configmap/secret's data.
+func determineStrategy(olds, news []provider.Manifest, workloadRefs []config.K8sResourceReference, logger *zap.Logger) (strategy model.SyncStrategy, summary string) {
```

Comment:
📝 It seems to be based on the current logic `decideStrategy`
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/planner/kubernetes/kubernetes.go#L233